src/main/java/org/logstash/input/DeadLetterQueueInputPlugin.java

```diff
@@ -34,7 +34,7 @@
 import java.util.function.Consumer;
 
 
-public class DeadLetterQueueInputPlugin {
+public class DeadLetterQueueInputPlugin implements AutoCloseable{
```

Comment:
I don't have strong feelings either way - I added the AutoCloseable interface to allow the use of try.. with resources to help avoid resource leaks/clean up tests. My understanding was the AutoCloseable was more used for code targeting JDK7+, but I guess you are right in that the existing close() throws IOException implies Closeable. (That also means that I need to fix up the method to make it idempotent, which it should have been anyway... ;))